gitindex/index.go

```diff
@@ -532,6 +518,225 @@ func newIgnoreMatcher(tree *object.Tree) (*ignore.Matcher, error) {
 	return ignore.ParseIgnoreFile(strings.NewReader(content))
 }
 
+func prepareDeltaBuild(options Options, repository *git.Repository) (repos map[fileKey]BlobLocation, branchMap map[fileKey][]string, branchVersions map[string]map[string]plumbing.Hash, changedOrDeletedPaths []string, err error) {
```

Comment:
phew this is all quite tricky! nice work getting something working. I have spent quite a bit of time trying to absorb this and give it a proper review but it is difficult. I think when we pair again it would be great to work on this code. I'd like to gain a deeper understanding since there seems like there are a lot of subtleties  to get right.

Reply:
I'm happy to take a look at this with you in person, but is there anything that jumps out at you as not-ideal that's impacting how grokkable this code is? (poor organization, etc.)

Reply:
It was just dense, which isn't a criticism of your code but really just of how this bit of zoekt-git-index has always been. There is likely ways we can rearrange how this is all done (for delta and non-delta) to make this clearer.